iocore/hostdb/HostDB.cc

```diff
@@ -659,6 +660,21 @@ db_mark_for(IpAddr const &ip)
 HostDBInfo *
 probe(ProxyMutex *mutex, HostDBMD5 const &md5, bool ignore_timeout)
 {
+  // If we have an entry in our hosts file, we don't need to bother with DNS
+  Ptr<RefCountedHostsFileMap> current_host_file_map = hostDB.hosts_file_ptr;
+
+  ts::ConstBuffer hname(md5.host_name, md5.host_len);
+  std::_Rb_tree_iterator<std::pair<const ts::ConstBuffer, IpAddr> > find_result = current_host_file_map->hosts_file_map.find(hname);
```

Comment:
It's not a good idea to dig around in the internals of `std::map`. The official way to do this is `std::map<ts::ConstBuffer, IpAddr, CmpConstBufferCaseInsensitive>::iterator`.